ui/src/lib/helpers.tsx

```diff
@@ -282,3 +282,17 @@ export const checkLinkIsValid = (text: string) => {
     );
     return urlR.test(lowerCaseText);
 };
+
+export const linkedPublicationTypes = (publicationType: Types.PublicationType): string[] => {
+    const types = {
+        PROBLEM: ['PROBLEM', 'HYPOTHESIS'],
+        HYPOTHESIS: ['PROBLEM', 'PROTOCOL'],
+        PROTOCOL: ['PROBLEM', 'DATA'],
+        DATA: ['PROBLEM', 'ANALYSIS'],
+        ANALYSIS: ['PROBLEM', 'INTERPRETATION'],
+        INTERPRETATION: ['PROBLEM', 'REAL_WORLD_APPLICATION'],
+        REAL_WORLD_APPLICATION: ['PROBLEM']
+    };
+
+    return types[publicationType as keyof typeof types];
```

Comment:
Your first attempt with the linkedPublicationTypes as an object was better. You just needed to provide the type for it.

For example:

export const linkedPublicationTypes: { [key: string]: Types.PublicationType[] } = {
        PROBLEM: ['PROBLEM', 'HYPOTHESIS'],
        HYPOTHESIS: ['PROBLEM', 'PROTOCOL'],
        // ...rest of them
};

And then in SidebarCard/Actions: Helpers.linkedPublicationTypes[props.publication.type].map(...)  will work as expected.

Reply:
Ah ok - I changed it because I noticed that's how the `helpers.formatPublicationType` does it and wanted to keep it in line with that, just incase we need to reuse the linked publication types helper somewhere else in the code then there's no confusion. Shall I change it back to how it was?

Reply:
Oh, I see. Then you can change the type of types and simply return types[publicationType];

Reply:
Ah ok - good point! Thanks - I've changed it back to the object :)